notes/download.py

```diff
@@ -81,6 +93,9 @@ def preprocess_notes(notes: pd.DataFrame) -> pd.DataFrame:
     # and rename some for simplicity
     notes = notes.rename(columns=cols, errors="ignore")
 
+    # drop the last_note_id if it was included
+    notes = notes.drop(index=notes.loc[notes["id_note"].eq(last_note_id)].index)
```

Comment:
This is the main change right here, using `last_note_id` to drop any record with a matching `id_note` column.